src/api/python/hidra/utils/utils_general.py

```diff
@@ -371,6 +372,35 @@ def create_sub_dirs(dir_path, subdirs, dirs_not_to_create=()):
         raise OSError
 
 
+class TempFile:
+    def __init__(self, temp_file, target_path):
+        self.temp_file = temp_file
+        self.target_path = str(target_path)
+
+    def seek(self, offset, whence=0):
+        return self.temp_file.seek(offset, whence)
+
+    def truncate(self, size=None):
+        return self.temp_file.truncate(size)
+
+    def write(self, data):
+        return self.temp_file.write(data)
+
+    def close(self):
+        if not self.temp_file.closed:
+            self.temp_file.close()
+            os.chmod(self.temp_file.name, 0o644)
+            os.rename(self.temp_file.name, self.target_path)
+
+
+def open_tempfile(filepath):
+    filepath = Path(filepath)
+    f = tempfile.NamedTemporaryFile(
+        suffix='.tmp', prefix=filepath.name + '.', dir=str(filepath.parent),
+        delete=False)
```

Comment:
Would `delete=False` help here, since the file is anyway renamed on `close()`? I assume the idea is to avoid multiple close events on `target_path`, so `temp_file` would need to be closed only once? Would `delete=True` be unsafe?

Reply:
`close` can be called multiple times but only has an effect the first time. The implementation tries to copy the behavior of normal Python file objects.

With `delete=True` (the default), the `NamedTemporaryFile` would be automatically deleted on close and the data would be lost.